pkg/authorization/registry/rolebinding/policybased/virtual_storage.go

```diff
@@ -158,59 +165,87 @@ func (m *VirtualStorage) UpdateRoleBindingWithEscalation(ctx kapi.Context, obj *
 }
 
 func (m *VirtualStorage) updateRoleBinding(ctx kapi.Context, name string, objInfo rest.UpdatedObjectInfo, allowEscalation bool) (*authorizationapi.RoleBinding, bool, error) {
-	old, err := m.Get(ctx, name)
-	if err != nil {
-		return nil, false, err
-	}
+	var updatedRoleBinding *authorizationapi.RoleBinding
+	var roleBindingConflicted = false
 
-	obj, err := objInfo.UpdatedObject(ctx, old)
-	if err != nil {
-		return nil, false, err
-	}
+	if err := kclient.RetryOnConflict(kclient.DefaultRetry, func() error {
+		old, err := m.Get(ctx, name)
+		if err != nil {
+			return err
+		}
+		oldRoleBinding, ok := old.(*authorizationapi.RoleBinding)
+		if !ok {
+			return kapierrors.NewBadRequest(fmt.Sprintf("old obj is not a role binding: %#v", oldRoleBinding))
+		}
 
-	roleBinding, ok := obj.(*authorizationapi.RoleBinding)
-	if !ok {
-		return nil, false, kapierrors.NewBadRequest(fmt.Sprintf("obj is not a role: %#v", obj))
-	}
+		obj, err := objInfo.UpdatedObject(ctx, old)
+		if err != nil {
+			return err
+		}
 
-	if err := rest.BeforeUpdate(m.UpdateStrategy, ctx, obj, old); err != nil {
-		return nil, false, err
-	}
+		roleBinding, ok := obj.(*authorizationapi.RoleBinding)
+		if !ok {
+			return kapierrors.NewBadRequest(fmt.Sprintf("obj is not a role binding: %#v", obj))
+		}
 
-	if err := m.validateReferentialIntegrity(ctx, roleBinding); err != nil {
-		return nil, false, err
-	}
-	if !allowEscalation {
-		if err := m.confirmNoEscalation(ctx, roleBinding); err != nil {
-			return nil, false, err
+		if len(roleBinding.ResourceVersion) == 0 && m.UpdateStrategy.AllowUnconditionalUpdate() {
+			roleBinding.ResourceVersion = oldRoleBinding.ResourceVersion
 		}
-	}
 
-	policyBinding, err := m.getPolicyBindingForPolicy(ctx, roleBinding.RoleRef.Namespace, allowEscalation)
-	if err != nil {
-		return nil, false, err
-	}
+		if err := rest.BeforeUpdate(m.UpdateStrategy, ctx, obj, old); err != nil {
+			return err
+		}
 
-	previousRoleBinding, exists := policyBinding.RoleBindings[roleBinding.Name]
-	if !exists {
-		return nil, false, kapierrors.NewNotFound(authorizationapi.Resource("rolebinding"), roleBinding.Name)
-	}
-	if previousRoleBinding.RoleRef != roleBinding.RoleRef {
-		return nil, false, errors.New("roleBinding.RoleRef may not be modified")
-	}
+		if err := m.validateReferentialIntegrity(ctx, roleBinding); err != nil {
```

Comment:
Pre-existing, but this check is only required if you're going to check for escalations.  Otherwise, you should be able to reference any one you want, right?  No need to race ourselves here.